homeassistant/components/homekit/type_covers.py

```diff
@@ -96,3 +98,63 @@ def update_state(self, new_state):
                     abs(current_position - self.homekit_target) < 6:
                 self.char_target_position.set_value(current_position)
                 self.homekit_target = None
+
+
+@TYPES.register('WindowCoveringBasic')
+class WindowCoveringBasic(HomeAccessory):
+    """Generate a Window accessory for a cover entity.
+
+    The cover entity must support:
+        - open_cover
+        - close_cover
+        - stop_cover (optional)
+    """
+
+    def __init__(self, *args, config):
+        """Initialize a WindowCovering accessory object."""
+        super().__init__(*args, category=CATEGORY_WINDOW_COVERING)
+        features = self.hass.states.get(self.entity_id) \
+            .attributes.get(ATTR_SUPPORTED_FEATURES)
+        self.supports_stop = features & SUPPORT_STOP
+
+        serv_cover = add_preload_service(self, SERV_WINDOW_COVERING)
+        self.char_current_position = setup_char(
+            CHAR_CURRENT_POSITION, serv_cover, value=0)
+        self.char_target_position = setup_char(
+            CHAR_TARGET_POSITION, serv_cover, value=0,
+            callback=self.move_cover)
+
+    def move_cover(self, value):
+        """Move cover to value if call came from HomeKit."""
+        _LOGGER.debug('%s: Set position to %d', self.entity_id, value)
+
+        service, position = (None, None)
```

Comment:
This is probably not necessary, since service and position are defined for every case.

Reply:
Same as below RE static analysis tools

Reply:
In that case, you might as well leave it in there ;)